src/huggingface_hub/repository.py

```diff
@@ -981,6 +1002,49 @@ def lfs_enable_largefiles(self):
         except subprocess.CalledProcessError as exc:
             raise EnvironmentError(exc.stderr)
 
+    def auto_track_binary_files(self, pattern: Optional[str] = ".") -> List[str]:
```

Comment:
this probably needs to be a private method

```suggestion
    def _auto_track_binary_files(self, pattern: Optional[str] = ".") -> List[str]:
```

Reply:
This should actually be a public method, same as `auto_track_large_files`. I think it's viable for a user to do `repo.auto_track_large_files` and `auto_track_binary_files` independently of `git_add`!